framework/sandstone.cpp

```diff
@@ -3589,6 +3621,90 @@ int main(int argc, char **argv)
         }
     }
 
+    /* initialize global lookup (all known tests) depending on the mode we
+     * started in */
+    SandstoneTestSet::init((sApp->shmem->selftest) ? SandstoneTestSet::SELF_TESTS : SandstoneTestSet::REGULAR_TESTS);
+
+    if (enabled_tests.size() || builtin_test_list_name || test_list_file_path) {
+        /* if anything other than the "all tests" has been specified, start with
+         * an empty list. */
+        test_set = new SandstoneTestSet(SandstoneTestSet::NO_TESTS);
+    } else {
+        /* otherwise, start with all the applicable tests (self tests or
+         * regular. */
+        test_set = new SandstoneTestSet((sApp->shmem->selftest) ? SandstoneTestSet::SELF_TESTS : SandstoneTestSet::REGULAR_TESTS);
```

Comment:
Sounds redundant with the `init` call above. Any way to simplify?

Reply:
`init` creates a catalog of all known tests, constructor is used to creates a instance which contains the tests to be run in this execution of opendcdiag. That's the difference. The reason it's not simply boolean in the constructor is that we need to start with an empty set at times (e.g. when `-e` is specified on the command line).